File: mobile/lib/models/reservation.dart
class Reservation {
  final int id;
  final int userId;
  final int spotId;
  final String? plateNumber;
  final String status;
  final DateTime startTime;
  final DateTime endTime;
  final DateTime? actualEntryTime;
  final DateTime? actualExitTime;
  final double? totalAmount;
  final DateTime createdAt;

  Reservation({
    required this.id,
    required this.userId,
    required this.spotId,
    this.plateNumber,
    required this.status,
    required this.startTime,
    required this.endTime,
    this.actualEntryTime,
    this.actualExitTime,
    this.totalAmount,
    required this.createdAt,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      id: json['id'],
      userId: json['user_id'],
      spotId: json['spot_id'],
      plateNumber: json['plate_number'],
      status: json['status'],
      startTime: DateTime.parse(json['start_time']),
      endTime: DateTime.parse(json['end_time']),
      actualEntryTime: json['actual_entry_time'] != null ? DateTime.parse(json['actual_entry_time']) : null,
      actualExitTime: json['actual_exit_time'] != null ? DateTime.parse(json['actual_exit_time']) : null,
      totalAmount: json['total_amount']?.toDouble(),
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  bool get isActive => status == 'active';
  bool get isCancelled => status == 'cancelled';
  bool get isCompleted => status == 'completed';
}

File: backend/app/models/reservation.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship
from app.database import Base


class Reservation(Base):
    __tablename__ = "reservations"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    vehicle_id = Column(Integer, ForeignKey("vehicles.id"), nullable=True)
    spot_id = Column(Integer, ForeignKey("parking_spots.id"), nullable=False)
    plate_number = Column(String(20), nullable=True)
    status = Column(String(20), default="active")  # active, completed, cancelled, no_show
    start_time = Column(DateTime(timezone=True), nullable=False)
    end_time = Column(DateTime(timezone=True), nullable=False)
    actual_entry_time = Column(DateTime(timezone=True), nullable=True)
    actual_exit_time = Column(DateTime(timezone=True), nullable=True)
    total_amount = Column(Float, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

    user = relationship("User", back_populates="reservations")
    vehicle = relationship("Vehicle", back_populates="reservations")
    spot = relationship("ParkingSpot", back_populates="reservations")
